File: Tokens/PlaylistTypeToken.cs
using iTunesLib;

namespace iTunesRichPresence_Rewrite.Tokens {
    public class PlaylistTypeToken : IToken {
        public string DisplayName => "Playlist type";
        public string Token => "%playlist_type";
        public string GetText(IiTunes iTunes) {
            if (iTunes.CurrentPlaylist.Kind == ITPlaylistKind.ITPlaylistKindUser) {
                return ((IITUserPlaylist) iTunes.CurrentPlaylist).SpecialKind == ITUserPlaylistSpecialKind.ITUserPlaylistSpecialKindMusic ? "Album" : "Playlist";
            }

            return "Album";
        }
    }
}

File: src/main/templating/tokens/PlaylistTypeToken.ts
import { Token } from '../types';
import { CurrentSong, PlaylistKind, PlaylistSpecialKind } from '../../itunes/types';

export default class PlaylistTypeToken implements Token {
    token = 'playlist_type';
    displayName = 'PlaylistType';

    render(currentSong: CurrentSong): string {
        if (
            currentSong.Playlist.Kind === PlaylistKind.ITPlaylistKindUser &&
            currentSong.Playlist.SpecialKind !== PlaylistSpecialKind.ITUserPlaylistSpecialKindMusic
        ) {
            return 'Music';
        }
        return 'Album';
    }
}
